libraries/core-react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,235 @@
+            onChange={onDateValueChange}
+            dateFormat={dateFormat}
+            placeholderText={placeholder}
+            id={id}
+            filterDate={(date: Date): boolean => {
+              if (disableFuture) {
+                return new Date() > date
+              }
+              if (disableBeforeDate) {
+                return date >= disableBeforeDate
+              }
+              return true
+            }}
+            onChangeRaw={(): void => {
+              ref?.setOpen?.(false)
+            }}
+            autoComplete="false"
+            popperPlacement={popperPlacement}
+            renderCustomHeader={(props) => (
+              <PopupHeader {...props} changeDate={onDateValueChange} />
+            )}
+            shouldCloseOnSelect={true}
+            readOnly={readOnly}
+          />
+          <CalendarIcon
+            name="calendar"
+            className="calendar-icon"
+            data={calendar}
+            size={24}
+          />
+        </DateLabel>
+      </Container>
+    )
+  },
+)
+console.log('tokens ---> ', tokens)
+
+const Container = styled.div.attrs<DatePickerToken>(tokens)`
+  width: 100%;
+  max-width: 148px;
+  display: flex;
+  flex-direction: column;
+  position: relative;
+  box-sizing: border-box;
+
+  .react-datepicker {
+    font-family: ${tokens.entities.title.typography.fontFamily} !important;
+  }
+  .react-datepicker__month-container {
+    -width: ${tokens.width};
+  }
+  .react-datepicker__header {
+    background-color: ${tokens.background};
+    border-bottom: none;
+  }
+  .react-datepicker__current-month {
+    font-weight: 500;
+  }
+  .react-datepicker__day-names {
+    border-top: 1px solid #ececec;
+    margin: 10px 0 0 0;
+    padding: 10px 0 0 0;
+    text-align: center;
+  }
+  .react-datepicker__day-name {
+    color: #6f6f6f;
+    font-size: 11px;
+    visibility: hidden;
+    line-height: unset;
+    width: 35px;
+  }
+  .react-datepicker__day-name:first-letter {
+    visibility: visible;
+  }
+  .react-datepicker__day {
+    color: #007079;
+    font-weight: 500;
+    width: 35px;
+    line-height: 35px;
+  }
+  .react-datepicker__day--selected,
+  .react-datepicker__month-text--keyboard-selected,
+  .react-datepicker__quarter-text--keyboard-selected,
+  .react-datepicker__year-text--keyboard-selected {
+    background-color: #007079;
+    color: white;
+  }
+  .react-datepicker__day--keyboard-selected {
+    background-color: unset;
+  }
+  .react-datepicker__day--disabled {
+    color: #ccc;
+  }
+  .react-datepicker-popper {
+    z-index: 4 !important;
+  }
+`
+
+const DateLabel = styled.label`
+  position: relative;
+  color: ${tokens.entities.label.typography.color};
+  font-family: ${tokens.entities.label.typography.fontFamily};
+  font-size: ${tokens.entities.label.typography.fontSize};
+  line-height: ${tokens.entities.label.typography.lineHeight};
+
+  span {
+    padding-left: 8px;
+  }
+`
+
+const StyledDatepicker = styled(DatePicker)`
```

Comment:
Maybe try the [customInput](https://reactdatepicker.com/#example-custom-input) with our `Input` as there are some styling faults with this one. Saves you the trouble of trying to recreate the existing one :)